src/convenience/webhook.ts

```diff
@@ -23,6 +23,53 @@ const callbackAdapter: FrameworkAdapter = (
 });
 const adapters = { ...nativeAdapters, callback: callbackAdapter };
 
+/**
+ * Performs a constant-time comparison of two strings to prevent timing attacks.
+ * This function always compares all bytes regardless of early differences,
+ * ensuring the comparison time does not leak information about the secret.
+ *
+ * @param header The header value from the request (X-Telegram-Bot-Api-Secret-Token)
+ * @param token The expected secret token configured for the webhook
+ * @returns true if strings are equal, false otherwise
+ */
+function compareSecretToken(
+    header: string | undefined,
+    token: string | undefined,
+): boolean {
+    // If no token is configured, accept all requests
+    if (token === undefined) {
+        return true;
+    }
+    // If token is configured but no header provided, reject
+    if (header === undefined) {
+        return false;
+    }
+
+    // Convert strings to Uint8Array for byte-by-byte comparison
+    const encoder = new TextEncoder();
+    const headerBytes = encoder.encode(header);
+    const tokenBytes = encoder.encode(token);
+
+    // Check if lengths differ (1 if different, 0 if same)
+    let hasDifference = headerBytes.length !== tokenBytes.length ? 1 : 0;
+
```

Comment:
There is nothing to be gained from keeping the token length a secret. We can reject early.
```suggestion
    // If lengths differ, reject
    if (headerBytes.length !== tokenBytes.length) {
      return false;
    }

    let hasDifference = 0;
```